src/agoradatatools/process.py

```diff
@@ -193,33 +197,36 @@ def process_all_files(
     if datasets:
         for dataset in datasets:
             try:
-                process_dataset(dataset_obj=dataset, staging_path=staging_path, syn=syn)
+                process_dataset(
+                    dataset_obj=dataset,
+                    staging_path=staging_path,
+                    syn=syn,
+                )
             except Exception as e:
                 error_list.append(
                     f"{list(dataset.keys())[0]}: " + str(e).replace("\n", "")
                 )
 
     destination = config["destination"]
 
-    if not error_list:
-        # create manifest if there are no errors
-        manifest_df = create_data_manifest(syn=syn, parent=destination)
-        manifest_path = load.df_to_csv(
-            df=manifest_df, staging_path=staging_path, filename="data_manifest.csv"
-        )
-
-        load.load(
-            file_path=manifest_path,
-            provenance=manifest_df["id"].to_list(),
-            destination=destination,
-            syn=syn,
-        )
-    else:
+    if error_list:
         raise ADTDataProcessingError(
             "\nData Processing has failed for one or more data sources. Refer to the list of errors below to address issues:\n"
             + "\n".join(error_list)
         )
 
+    manifest_df = create_data_manifest(syn=syn, parent=destination)
```

Comment:
Does the manifest now get written to Synapse when there is a processing error?

Reply:
No, because if there are errors `ADTDataProcessingError` is raised, resulting in the process being terminated there.  I un-nested the code because it's cleaner.